scripts/openmic_split.py

```diff
@@ -84,13 +104,55 @@ def load_label_matrix(metadata_file, label_file):
                                    values='relevance',
                                    index='index')
 
+    artist_labels = pd.merge(label_matrix, skm[['artist_id']],
+                             left_index=True,
+                             right_index=True,
+                             how='right').groupby('artist_id').mean()
+
     # And fill in an extra column for all-negative examples
-    label_matrix['_negative'] = (label_matrix.max(axis=1) < 0) * 1.0
+    artist_labels['_negative'] = (artist_labels.max(axis=1) < 0) * 1.0
+
+    label_matrix.index = meta['sample_key']
+
+    return skey, artist_labels, label_matrix
+
+
+def check_prob(label_matrix, idx, prob_ratio):
+    '''Check that the probabilities in a sub-sample
+    are within a tolerance of the full population.
+
+    Parameters
+    ----------
+    label_matrix : pd.DataFrame
+        Array of label assignments
 
-    return skey['sample_key'], label_matrix
+    idx : iterable
+        Indices of the target sub-sample
 
+    prob_ratio:
+        The target probability ratio
 
-def make_partitions(metadata, labels, seed, num_splits, ratio):
+    Returns
+    -------
+    check_passed : bool
+        True if the sub-sampled distribution is within tolerance
+        False otherwise
+    '''
+    min_prob, max_prob = sorted([prob_ratio, 1./prob_ratio])
+
+    all_dist_p = (label_matrix > 0).sum() / label_matrix.count()
+    all_dist_n = (label_matrix <= 0).sum() / label_matrix.count()
+
+    sub_dist_p = (label_matrix.loc[idx] > 0).sum() / label_matrix.loc[idx].count()
+    sub_dist_n = (label_matrix.loc[idx] <= 0).sum() / label_matrix.loc[idx].count()
+
+    return (np.all(min_prob * all_dist_p.values <= sub_dist_p.values) and
```

Comment:
not sure if this is the best place to do so, but looking at this logic for the first time / without a lot of context, I'm having trouble unpacking the intuition here. Would an inline comment make sense? Or in the docstring?

Reply:
Yeah, a comment makes sense here.  I'll add it.